test/cppsuite/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+project(cppsuite CXX)
+
+create_test_executable(run
+    SOURCES tests/run.cxx
+    FLAGS -std=c++11
+    INCLUDES ${CMAKE_CURRENT_SOURCE_DIR}
+    ADDITIONAL_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/configs
+    CXX
```

Comment:
I guess the part I'm referring to is probably already merged in, but how do we decide whether to build C++ stuff or not? Is it the same thing it used to be where we check whether `CXX == c++`?

Reply:
Good point, CMake figures out whether its building C or C++ through the header in the build file e.g `project(cppsuite CXX)` (for C++, in this file) or `project(csuite C)` (for C code). This tells it what toolchain to use for the executable's and libraries defined under the project. We don't really need to do anything else.

I added this specific `CXX` option to the helper since the function by default appends diagnostic flags that may have been generated due to the strict option (Seen here: https://github.com/wiredtiger/wiredtiger/blob/a102c85ab0b50b6f56791512b27acf602e468aef/test/ctest_helpers.cmake#L83). This function was mainly developed for our C tests, however I wanted to take advantage of the helper for the `cppsuite` since its mostly the same logic.
 
I could expand the helper to be a bit more aware of the language its building, as to avoid passing a `CXX` option? (though it didn't seem we are building much C++ outside of this suite).

Reply:
Sorry, my question was pretty vague. In the AutoTools config, there was some rule that was something along the lines of: if we choose a C compiler that it something other than our system's `cc`, we don't build C++ code.

I think this is the relevant bit: https://github.com/wiredtiger/wiredtiger/blob/develop/build_posix/configure.ac.in#L28

Does it work the same way in CMake? I don't really have an opinion on how it should work, I was just curious.

Reply:
Oh I wasn't aware of that check. CMake currently defaults to `cc` if no specific compiler was passed in though we currently don't limit C++ compilation to only the systems `cc`. CMake defines the C++ compiler as `CMAKE_CXX_COMPILER`. This being independent from the C compiler definition `CMAKE_C_COMPILER`.  

Those variables can be independently overwritten to any compilers, the Clang toolchain file (https://github.com/wiredtiger/wiredtiger/blob/develop/build_cmake/toolchains/clang.cmake) and GCC toolchain files (https://github.com/wiredtiger/wiredtiger/blob/develop/build_cmake/toolchains/gcc.cmake) are examples of manually setting the C compiler and C++ compiler to something that is not `cc`. 

Not sure if we want to add this restriction into CMake? 
CMake already checks if the given C++ compiler works during the configuration process (can't set `CMAKE_CXX_COMPILER` with `my_totally_broken_c++_compiler`). It checks the C++ compilers ABI info, compiler features and if it can perform a simple C++ compilation. It will throw an error if its an invalid C++ compiler.

Reply:
Cool. That seems like more sensible behavior anyway.